src/main/java/org/openstreetmap/atlas/geography/atlas/items/ConnectedEdgeType.java

```diff
@@ -0,0 +1,26 @@
+package org.openstreetmap.atlas.geography.atlas.items;
+
+import java.util.SortedSet;
+import java.util.function.Function;
+
+import lombok.AllArgsConstructor;
+import lombok.Getter;
+
+/**
+ * {@link Node}s have in and out edges, this enum enables generalizing logic around this node and
+ * edge connection.
+ *
+ * @author Yazad Khambata
+ */
+@AllArgsConstructor
+@Getter
+public enum ConnectedEdgeType
```

Comment:
This is a great idea. I may be able to update some code down the line.

Reply:
Go for it :)